test/test_common/network_utility.h

```diff
@@ -33,6 +33,13 @@ Address::InstanceConstSharedPtr findOrCheckFreePort(const std::string& addr_port
                                                     Address::SocketType type);
 
 /**
+ * Get IP loopback address as a string.
```

Comment:
Technically the address is "::1", and "[::1]" is a special case used in URL because : was already used as the port delimiter and the IPv6 address shortening scheme (double ::) means an URL parser can't figure out whether the final : is a port delimiter or just part of the address without some way of surrounding the address.

If the purpose of this method is to return an "URL ready" formatted address I think the name should be changed to indicate this.  Otherwise maybe it should be the caller's responsibility to add the [ ] when creating the URL in a fmt::format.